File: SoftUni - C# OOP/Inheritance/Inheritance - Exercise/04. NeedForSpeed/RaceMotorcycle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NeedForSpeed
{
    public class RaceMotorcycle : Motorcycle
    {
        public RaceMotorcycle(int horsePower, double fuel) : base(horsePower, fuel)
        {

        }
        public override void Drive(double kilometers)
        {
            DefaultFuelConsumption = 8;
            //this.Fuel -= this.FuelConsumption = 8 * kilometers;
            this.Fuel -= DefaultFuelConsumption * kilometers;

        }
    }
}

File: SoftUni - Java OOP/Inheritance/Inheritance - Exercises/_04_Need_For_Speed/RaceMotorcycle.java
package _04_Need_For_Speed;

public class RaceMotorcycle extends Motorcycle {
    private static final double DEFAULT_FUEL_CONSUMPTION = 8;
    public RaceMotorcycle(double fuel, int horsePower) {
        super(fuel, horsePower);
        super.setFuelConsumption(DEFAULT_FUEL_CONSUMPTION);
    }
}
